app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,36 @@ def show
     @post = Post.find(params[:id])
     @comments = @post.comments.all
   end
+
+  def new
+    @user = current_user
+  end
+
+  def create
+    @user = current_user
+    p(post_params)
+    @post = @user.posts.new(post_params)
+    p(@post)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def like
+    @post = Post.all.find(params[:id])
```

Comment:
**_[OPTIONAL]_** 

`Post.all.find(params[:id])` 
is the same as 
`Post.find(params[:id])`

- [OPTIONAL] Change `Post.all` to `Post`